src/gitlab.js

```diff
@@ -32,68 +32,85 @@ export default class VerdaccioGitLab {
 
   }
 
+  static get logPrefix() { return '[gitlab]'; }
+
   authenticate(user: string, password: string, cb: Callback) {
+    this.logger.trace(VerdaccioGitLab.logPrefix, 'authenticate called for user:', user);
 
-    var GitlabAPI = new Gitlab({
+    const GitlabAPI = new Gitlab({
       url:   this.config.url,
       token: password
     });
 
-    GitlabAPI.Users.current().then((response) => {
-      if (user !== response.username) return cb(httperror[403]('wrong gitlab username'));
-      var ownedGroups = [user];
-      GitlabAPI.Groups.all({'owned': 'true'}).then((groups) => {
-        groups.forEach(function(item) {
-          if (item.path === item.full_path) {
-            ownedGroups.push(item.path);
+    GitlabAPI.Users.current().then(response => {
+      if (user !== response.username) { return cb(httperror[401]('wrong gitlab username')); }
+
+      // Set the groups of an authenticated user to themselves and all gitlab projects of which they are an owner
+      let ownedGroups = [user];
+      GitlabAPI.Groups.all({'owned': 'true'}).then(groups => {
+        for (let group of groups) {
+          if (group.path === group.full_path) {
+            ownedGroups.push(group.path);
           }
-        });
+        }
+
+        this.logger.debug(VerdaccioGitLab.logPrefix, 'user:', user, 'authenticated, with groups:', ownedGroups);
         cb(null, ownedGroups);
       });
     }).
     catch(error => {
-      if (error) return cb('Personal access token invalid');
+      this.logger.debug(VerdaccioGitLab.logPrefix, 'error authenticating:', error.error || null);
+      if (error) { return cb(httperror[401]('personal access token invalid')); }
     });
   }
 
   adduser(user: string, password: string, cb: Callback) {
-    cb(null, false);
+    this.logger.trace(VerdaccioGitLab.logPrefix, 'adduser called for user:', user);
+    cb(null, true);
```

Comment:
This needs to be changed to true so that the authentication function is called when trying to login. Otherwise all logins succeed at the npm client side